VMEncryption/main/BekUtil.py

```diff
@@ -16,151 +16,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from Common import TestHooks, CommonVariables
-import base64
-import os.path
-import traceback
-import sys
-
-"""
-add retry-logic to the network api call.
-"""
-class BekMissingException(Exception):
-
-   def __init__(self, value):
-      self.value = value
-
-   def __str__(self):
-      return(repr(self.value))
+from Common import CommonVariables
+from IMDSUtil import IMDSStoredResults
+from BekUtilVolumeImpl import BekUtilVolumeImpl
+from BekUtilFileImpl import BekUtilFileImpl
 
 class BekUtil(object):
     """
     Utility functions related to the BEK VOLUME and BEK files
     """
-    def __init__(self, disk_util, logger):
-        self.disk_util = disk_util
-        self.logger = logger
-        self.bek_filesystem_mount_point = '/mnt/azure_bek_disk'
-        self.bek_label = 'BEK VOLUME'
-        self.bek_filesystem = 'vfat'
-        self.wrong_fs_msg = "BEK does not have vfat filesystem."
-        self.not_mounted_msg = "BEK is not mounted."
-        self.partition_missing_msg = "BEK disk does not expected partition."
-        self.bek_missing_msg = "BEK disk is not attached."
+    def __init__(self, disk_util, logger, encryption_environment=None):
+        security_type = CommonVariables.Standard
+        try:
+            imds_Stored_Results=IMDSStoredResults(logger=logger,encryption_environment=encryption_environment)
+            security_type = imds_Stored_Results.get_security_type()
+        except Exception as ex:
+            logger.log("Failure in retriving security type from IMDS, Following BEK Volume Path.")
+        finally:
+            logger.log("BEK util path is followed according to VM security type {0}".format(security_type))
+                
+        if security_type != CommonVariables.ConfidentialVM:
+           self.bekUtilImpl = BekUtilVolumeImpl(disk_util,logger)
+        else:
+            self.bekUtilImpl = BekUtilFileImpl(disk_util,logger)
```

Comment:
I don't understand as to why this was refactored into two implementations. Are these implementations different depending on the security type? If this is the case I think we should have a base class as BekUtil and a subclass called CVMBekUtil that overrides any behavior for CVM feature.

Reply:
Resolving based on our conversation during team meetings.

Reply:
For the record, this is for runtime polymorphism. Different implementation is chosen for CVM vs standard VM.